compiler/rustc_middle/src/traits/select.rs

```diff
@@ -12,12 +12,12 @@ use rustc_hir::def_id::DefId;
 use rustc_query_system::cache::Cache;
 
 pub type SelectionCache<'tcx> = Cache<
-    ty::ParamEnvAnd<'tcx, ty::TraitPredicate<'tcx>>,
+    (ty::ParamEnv<'tcx>, ty::TraitPredicate<'tcx>),
```

Comment:
Can you add comments here and below explaining why this is is *not* `ParamEnvAnd`?